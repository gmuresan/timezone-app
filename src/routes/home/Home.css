@import '../../components/variables.css';

.root {
  padding-left: 20px;
  padding-right: 20px;
}

.container {
  margin: 0 auto;
  padding: 0 0 40px;
  max-width: var(--max-content-width);
  display: flex;
  flex-direction: column;
}

.timezone {
  width: 300px;
  display: inline-block;
  margin-right: 15px;
}

.cardTopRow {
  display: flex;
  justify-content: space-between;
}

.topRow {
  display: flex;
}

