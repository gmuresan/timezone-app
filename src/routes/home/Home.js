import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import { Card, CardBlock, CardText, CardTitle, CardSubtitle } from 'reactstrap';
import moment from 'moment';

import s from './Home.css';
import ModalForm from '../../components/ModalForm';
import TimezoneForm from '../../forms/TimezoneForm';
import LoadingButton from '../../components/LoadingButton';
import Clock from '../../components/Clock';

class Home extends React.Component {
  static propTypes = {
    timezones: PropTypes.array.isRequired,
    createTimezone: PropTypes.func.isRequired,
    updateTimezone: PropTypes.func.isRequired,
    deleteTimezone: PropTypes.func.isRequired,
    filterByName: PropTypes.func.isRequired,
  };

  static defaultProps = {
    timezones: [],
  };

  constructor(props) {
    super(props);
    this.timezoneModal = ModalForm(TimezoneForm);

    this.state = {
      time: moment(),
    };
  }

  render() {
    const TzModal = this.timezoneModal;
    const timezones = this.props.timezones.map((tz) => {
      const time = moment(this.state.time);
      time.utcOffset(tz.gmtOffset);
      return (
        <div id={tz.id} className={s.timezone}>
          <Card>
            <CardBlock>
              <div className={s.cardTopRow}>
                <CardTitle>{tz.name}</CardTitle>
                <TzModal
                  onSubmit={(values) => this.props.updateTimezone(values, tz)}
                  buttonText="Edit"
                  title={`Edit ${tz.name}`}
                  values={tz}
                />
              </div>
              <CardSubtitle>City: {tz.city}</CardSubtitle>
              <CardText>GMT Offset: {tz.gmtOffset} hours</CardText>
              Current Time: <Clock gmtOffset={tz.gmtOffset} />
              <LoadingButton
                dirty
                text="Delete"
                loadingText="Deleting"
                successText="Deleted"
                onClick={() => this.props.deleteTimezone(tz)}
                red
              />

            </CardBlock>
          </Card>
        </div>
      );
    });

    return (
      <div className={s.root}>
        <div className={s.container}>
          <div>
            <h3>Timezones</h3>
            <div className={s.topRow}>
              <div>
                Search By Name: <input type="text" onChange={(e) => this.props.filterByName(e.target.value)} />
              </div>
              <this.timezoneModal
                onSubmit={this.props.createTimezone}
                buttonText="New Timezone"
                title="Create Timezone"
              />
            </div>
          </div>
          <div>
            {timezones}
          </div>

        </div>
      </div>
    );
  }
}

export default withStyles(s)(Home);
