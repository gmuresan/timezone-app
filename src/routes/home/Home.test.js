/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../components/App';
import Home from './Home';

const timezones = [
  { id: 1, name: 'n1', city: 'c1', gmtOffset: 1 },
  { id: 2, name: 'n2', city: 'c2', gmtOffset: 2 },
  { id: 3, name: 'n3', city: 'c3', gmtOffset: 3 },
];

describe('Home', () => {
  test('renders timezone list', () => {
    Date.now = jest.fn(() => 1487076708000);
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {} }}>
          <Home
            timezones={timezones}
            createTimezone={() => {}}
            updateTimezone={() => {}}
            deleteTimezone={() => {}}
            filterByName={() => {}}
          />
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});

