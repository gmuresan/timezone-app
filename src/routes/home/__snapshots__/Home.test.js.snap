// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders timezone list 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <div>
      <h3>
        Timezones
      </h3>
      <div
        className="topRow"
      >
        <div>
          Search By Name: 
          <input
            onChange={[Function]}
            type="text"
          />
        </div>
        <div>
          <div
            className="loadingButton"
            id="loading-button-wrapper-Save"
          >
            <button
              className="button    btn btn-warning"
              data-permission="admin"
              disabled={false}
              id="loading-button-Save"
              onClick={[Function]}
              type="button"
            >
              <div
                className="children"
              >
                New Timezone
              </div>
               
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="timezone"
      >
        <div
          className="card"
        >
          <div
            className="card-block"
          >
            <div
              className="cardTopRow"
            >
              <h4
                className="card-title"
              >
                n1
              </h4>
              <div>
                <div
                  className="loadingButton"
                  id="loading-button-wrapper-Save"
                >
                  <button
                    className="button    btn btn-warning"
                    data-permission="admin"
                    disabled={false}
                    id="loading-button-Save"
                    onClick={[Function]}
                    type="button"
                  >
                    <div
                      className="children"
                    >
                      Edit
                    </div>
                     
                  </button>
                </div>
              </div>
            </div>
            <h6
              className="card-subtitle"
            >
              City: 
              c1
            </h6>
            <p
              className="card-text"
            >
              GMT Offset: 
              1
               hours
            </p>
            Current Time: 
            <div>
              01:51:48 PM
               GMT
            </div>
            <div
              className="loadingButton"
              id="loading-button-wrapper-Delete"
            >
              <button
                className="button red   btn btn-warning"
                data-permission="admin"
                disabled={false}
                id="loading-button-Delete"
                onClick={[Function]}
                type="button"
              >
                Delete
                 
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="timezone"
      >
        <div
          className="card"
        >
          <div
            className="card-block"
          >
            <div
              className="cardTopRow"
            >
              <h4
                className="card-title"
              >
                n2
              </h4>
              <div>
                <div
                  className="loadingButton"
                  id="loading-button-wrapper-Save"
                >
                  <button
                    className="button    btn btn-warning"
                    data-permission="admin"
                    disabled={false}
                    id="loading-button-Save"
                    onClick={[Function]}
                    type="button"
                  >
                    <div
                      className="children"
                    >
                      Edit
                    </div>
                     
                  </button>
                </div>
              </div>
            </div>
            <h6
              className="card-subtitle"
            >
              City: 
              c2
            </h6>
            <p
              className="card-text"
            >
              GMT Offset: 
              2
               hours
            </p>
            Current Time: 
            <div>
              02:51:48 PM
               GMT
            </div>
            <div
              className="loadingButton"
              id="loading-button-wrapper-Delete"
            >
              <button
                className="button red   btn btn-warning"
                data-permission="admin"
                disabled={false}
                id="loading-button-Delete"
                onClick={[Function]}
                type="button"
              >
                Delete
                 
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="timezone"
      >
        <div
          className="card"
        >
          <div
            className="card-block"
          >
            <div
              className="cardTopRow"
            >
              <h4
                className="card-title"
              >
                n3
              </h4>
              <div>
                <div
                  className="loadingButton"
                  id="loading-button-wrapper-Save"
                >
                  <button
                    className="button    btn btn-warning"
                    data-permission="admin"
                    disabled={false}
                    id="loading-button-Save"
                    onClick={[Function]}
                    type="button"
                  >
                    <div
                      className="children"
                    >
                      Edit
                    </div>
                     
                  </button>
                </div>
              </div>
            </div>
            <h6
              className="card-subtitle"
            >
              City: 
              c3
            </h6>
            <p
              className="card-text"
            >
              GMT Offset: 
              3
               hours
            </p>
            Current Time: 
            <div>
              03:51:48 PM
               GMT
            </div>
            <div
              className="loadingButton"
              id="loading-button-wrapper-Delete"
            >
              <button
                className="button red   btn btn-warning"
                data-permission="admin"
                disabled={false}
                id="loading-button-Delete"
                onClick={[Function]}
                type="button"
              >
                Delete
                 
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
